### PR TITLE
11-delete-order and test case

### PR DESCRIPTION
**Pull request by kiara0-0**
An entire order can be deleted using an order id which will be passed in. 
The test case has also been written to check if an order which is created also gets deleted.

### CODE REVIEW

File: tests/test_routes.py [hunk 1]
```diff
@@ -140,3 +140,10 @@ def test_read_order(self):
         self.assertEqual(resp.status_code, status.HTTP_200_OK)
         data = resp.get_json()
         self.assertEqual(data["customer_name"], order.customer_name)
+
+    def test_delete_order(self):
+        """It should Delete an entire order by order id"""
+        order = self._create_orders(1)[0]
+        resp = self.client.delete(f"{BASE_URL}/{order.id}")
+        self.assertEqual(resp.status_code, status.HTTP_204_NO_CONTENT)
+        print("Success")
```
Comment (Hyxpillow) [inline tests/test_routes.py:149]: No need to print "Success" here

**Review by Hyxpillow — APPROVED** [review]
Looks good to me. There might be a conflict with Zhenghan Nan's PR #33  in test_models.py, but it's not a big issue

File: tests/test_models.py [hunk 2]
```diff
@@ -65,15 +65,15 @@ def tearDown(self):
...
-        """It should create a Order"""
-        # Todo: Remove this test case example
-        resource = OrderFactory()
-        resource.create()
-        self.assertIsNotNone(resource.id)
-        found = Order.all()
-        self.assertEqual(len(found), 1)
-        data = Order.find(resource.id)
-        self.assertEqual(data.name, resource.name)
+    # def test_example_replace_this(self):
```
Comment (anuva98) [inline tests/test_models.py:68]: Do we need this test case at all?

File: tests/test_routes.py [hunk 3]
```diff
@@ -25,7 +25,7 @@
 from wsgi import app
 from service.common import status
 from service.models import db, Order
-from tests.factories import OrderFactory, ItemFactory
```
Comment (anuva98) [inline tests/test_routes.py:28]: Why remove ItemFactory?

File: service/routes.py [hunk 4]
```diff
@@ -26,6 +26,8 @@
 from service.common import status  # HTTP Status Codes
 from service.models import Order
 
+# from .persistent_base import db
```
Comment (anuva98) [inline service/routes.py:29]: We don't need this here, can remove the line

**Review by anuva98 — CHANGES_REQUESTED** [review]

File: tests/test_models.py (hunk 2 above)
Comment (Hyxpillow) [reply tests/test_models.py:68]: Since you have added test_item.py and test_model.py in PR #35, we can discard test_models.py now

Comment (anuva98) [reply tests/test_models.py:68]: Ah yes, you are right. Completely slipped my mind. Shall I do it in the other one or let @kiara0-0 remove it in this one?

Comment (Hyxpillow) [reply tests/test_models.py:68]: It's up to you two~

Comment (anuva98) [reply tests/test_models.py:68]: Okay, I addressed this in my PR.. so no need to do it here then :)

File: tests/test_routes.py (hunk 1 above)
Comment (kiara0-0) [reply tests/test_routes.py:149]: Sorry, my bad, that was for debugging purposes

File: tests/test_routes.py (hunk 3 above)
Comment (kiara0-0) [reply tests/test_routes.py:28]: This was done because we removed the test case from this file, and so ItemFactory was not really used elsewhere

**Review by Hyxpillow — APPROVED** [review]
Looks good to me.

**Pull request closed without merging**